### ISSUE TITLE
User settings aren't merged

### ISSUE THREAD

**Issue by wesbos** [colinta/SublimeChangeQuotes#30]
If I try make a custom user setting in the Change quotes user file that changes one thing:

```js
{
  "lists": {
    "source.js": {
      "quotes": [["'", "\"", "`"]]
    }
  }
}
```

It looks like all the other settings from the default file are blown away, because it causes an error: `KeyError: 'default'`

If I copy/paste the entire default file into my user settings it works, but it should just add my new setting from User to the Default settings, no?

**Comment by colinta** [comment]
This is a limitation of Sublime - though, a way around this would be to hardcode the "defaults" in the python files, and merge in the user settings.

**Comment by pepkin88** [comment]
I think it could be bypassed by splitting the `lists` field to two fields: `default_lists` and `custom_lists`  (names are only an example). That way the user could overwrite some settings by copying some parts of `default_lists` to `custom_lists` and change it.
What do you think?

**Comment by colinta** [comment]
I'm not 100%, but I think that the way Sublime loads settings, it will *only* load from User or Default settings, never both.

**Comment by pepkin88** [comment]
From what I've seen in other plugins, it takes from both files, but it takes by key. So if the key exists in the user settings, it takes the whole value from there, if some key is absent, then it takes the value from the default settings.

**Comment by colinta** [comment]
Ahhh, right, that's true!

**Comment by pepkin88** [comment]
Although my solution would give us the control over merging the settings, it is quite unusual and unintuitive. It should be documented somewhere, preferably in the default settings file.

**Comment by wesbos** [comment]
Yeah I think a lot of sublime packages just fold the user settings overtop the defaults
